esmvalcore/preprocessor/_regrid.py

```diff
@@ -151,25 +192,95 @@ def _stock_cube(spec, lat_offset=True, lon_offset=True):
                               int(_LON_RANGE / dlon))
 
-    lats = iris.coords.DimCoord(
-        latdata,
-        standard_name='latitude',
-        units='degrees_north',
-        var_name='lat')
-    lats.guess_bounds()
+    cube = _stock_cube(latdata=latdata, londata=londata)
 
-    lons = iris.coords.DimCoord(
-        londata,
-        standard_name='longitude',
-        units='degrees_east',
-        var_name='lon')
-    lons.guess_bounds()
+    return cube
 
-    # Construct the resultant stock cube, with dummy data.
-    shape = (latdata.size, londata.size)
-    dummy = np.empty(shape, dtype=np.dtype('int8'))
-    coords_spec = [(lats, 0), (lons, 1)]
-    cube = iris.cube.Cube(dummy, dim_coords_and_dims=coords_spec)
+
+def _spec_to_latlonvals(*, lat_start: float, lat_end: float, lat_step: float,
+                        lon_start: float, lon_end: float,
+                        lon_step: float) -> tuple:
+    """Define lat/lon values from spec.
+
+    Create a regional cube starting defined by the target specification.
+
+    The latitude must be between -90 and +90. The longitude is not bounded, but
+    wraps around the full great circle.
+
+    Parameters
+    ----------
+    lat_start : float
+        Latitude value of the first grid cell center. The grid includes this
+        value.
+    lat_end : float
+        Maximum latitude value of the last grid cell. The grid includes this
+        value only if it falls on a grid point. Otherwise, it cuts off at the
+        first previous value.
+    lat_step : float
+        Latitude spacing between grid points, i.e. the distance between two
+        adjecent values.
+    lon_start : float
+        Latitude value of the first grid cell center. The grid includes this
+        value.
+    lon_end : float
+        Maximum longitude value of the last grid cell. The grid includes this
+        value only if it falls on a grid point. Otherwise, it cuts off at the
+        first previous value.
+    lon_step : float
+        Longitude spacing between grid points, i.e. the distance between two
+        adjecent values.
+
+    Returns
+    -------
+    xvals : np.array
+        List of longitudes
+    yvals : np.array
+        List of latitudes
+    """
+    if lat_step == 0:
+        raise ValueError('Latitude step must be larger than 0, '
+                         f'got lat_step={lat_step}.')
+
+    if lon_step == 0:
+        raise ValueError('Longitude step must be larger than 0, '
+                         f'got lon_step={lon_step}.')
+
+    if (lat_start < _LAT_MIN) or (lat_end > _LAT_MAX):
+        raise ValueError(
+            f'Latitude values must lie between {_LAT_MIN}:{_LAT_MAX}, '
+            f'got lat_start={lat_start}:lat_end={lat_end}.')
+
+    def get_points(start, stop, step):
+        """Calculate grid points."""
+        num = int((stop - start) // step)
+        stop = start + num * step
+        return np.linspace(start, stop, num + 1)
```

Comment:
don't like functions inside functions, can we take this outside?

Reply:
Normally, I would agree with you, but this function is very small and only relevant in this context that I'd prefer to keep it here for clarity. I did it this way to prevent repeating the same code twice for latitude/longitude.